### PR TITLE
feat: display collection profile in search result

### LINKED ISSUE #5500: Collection preview doesn't show in searchbar
### What happened?

doesn't shows collection profile

### Please reproduce in steps

search some collection like "Ku*" on rmrk2

### Expected Behavior

image

I would also change it to square picture as it's small wyt @exezbcz 

### What browsers are you seeing the problem on?

Chrome

### At which address did you encounter bug?

canary.kodadot.xyz

### Are you logged in?

None

### Which wallet you are using?

_No response_

### Screenshots

image

### Relevant log output

_No response_

### Payment link for reward

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by floyd-li**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5500 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EZiu1PjV2j2JHKxY6mHnFwwCRCoV27uHKQSkKXATSh1srJT)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### <samp>🤖 Generated by Copilot at 120792d</samp>

Improved the appearance and functionality of the search result components, showing larger and better-styled NFT thumbnails and displaying the floor price and the number of units for collections and series. Modified `SearchResultItem.vue` and `SearchSuggestion.vue` files.

### <samp>🤖 Generated by Copilot at 120792d</samp>

> _`NFT` thumbnails_
> _bigger, styled, informative_
> _showing floor price - spring_

### CODE REVIEW

File: components/search/SearchSuggestion.vue [hunk 1]
```diff
@@ -573,6 +586,16 @@ export default class SearchSuggestion extends mixins(PrefixMixin) {
...
   }
 
+  getFloorPrice(nfts) {
+    if (!nfts) {
+      return 0
+    }
+    return nfts
+      .map((x) => Number(x.price))
+      .sort()
+      .at(0)
```
Comment (daiagi) [inline components/search/SearchSuggestion.vue:596]:
```suggestion
return Math.min(...nft.map(nft => Number(nft.price)))
```

not efficient to use sort when you just want min

**Comment by yangwao** [comment]
Sometimes doesn't work hmm

image

I don't see any related errors 

image

File: components/search/SearchSuggestion.vue [hunk 2]
```diff
@@ -26,7 +26,20 @@
                 <div
                   class="is-flex is-flex-direction-row is-justify-content-space-between pt-2 pr-2">
                   <span class="main-title name">{{ item.name }}</span>
-                  <span>{{ urlPrefix.toUpperCase() }}</span>
+                  <span class="has-text-grey">
+                    {{ urlPrefix.toUpperCase() }}
+                  </span>
+                </div>
+                <div
+                  class="is-flex is-flex-direction-row is-justify-content-space-between pr-2">
```
Comment (daiagi) [inline components/search/SearchSuggestion.vue:34]:
```suggestion
                  class="is-flex is-justify-content-space-between pr-2">
```

flex-direction: row is default

**Review by daiagi — APPROVED** [review]
small thing
otherewise code lgtm :heavy_check_mark:

File: components/search/SearchSuggestion.vue [hunk 3]
```diff
@@ -573,6 +586,13 @@ export default class SearchSuggestion extends mixins(PrefixMixin) {
     }
   }
 
+  getFloorPrice(nfts) {
+    if (!nfts) {
+      return 0
+    }
+return Math.min(...nfts.map(nft => Number(nft.price)))
```
Comment (Jarsen136) [inline components/search/SearchSuggestion.vue:593]: floor price should be greater than zero.

**Review by Jarsen136 — CHANGES_REQUESTED** [review]

**Review by Jarsen136 — CHANGES_REQUESTED** [review]
- floor price should be greater than zero.
- ksm/bsx unit
image

**Comment by yangwao** [comment]
Any idea why it doesn't work for me? I don't see any error console nor pending network request in console. 

image

**Comment by yangwao** [comment]
@floyd-li it actually returns in network well, yet it doesn't shows?

My thesis is that.. if resources comes later than in some "first batch", we don't swap the new resource, is that possible?

image

**Comment by floyd-li** [comment]
> @floyd-li it actually returns in network well, yet it doesn't shows?
> 
> My thesis is that.. if resources comes later than in some "first batch", we don't swap the new resource, is that possible?
> 

working on this

**Comment by yangwao** [comment]
> working on this

looks like it's working! 😍

image

**Comment by Jarsen136** [comment]
Pls hide `floor: 0 KSM` where the collection has no item having a price.

Here is what opensea does in this case.
image

File: components/search/SearchSuggestion.vue (hunk 3 above)
Comment (Jarsen136) [reply components/search/SearchSuggestion.vue]: I think yes

**Comment by floyd-li** [comment]
image

is it okay?

**Comment by Jarsen136** [comment]
> image
> 
> is it okay?

Yes, it looks good.

**Review by Jarsen136 — APPROVED** [review]

**Comment by yangwao** [comment]
this was quick @floyd-li , love it!

pay 40 usd

let's look maybe on this one 

- https://github.com/kodadot/nft-gallery/issues/5725

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $40 @ 34.26 USD/KSM ~ 1.168 $KSM
🧗 [EZiu1PjV2j2JHKxY6mHnFwwCRCoV27uHKQSkKXATSh1srJT](https://kusama.subscan.io/account/EZiu1PjV2j2JHKxY6mHnFwwCRCoV27uHKQSkKXATSh1srJT)
🔗 [0x1ddbf6b06f6204ae1a2992bca766f4183d73b8e1fdf23815a1b3cbb29fcde35c](https://kusama.subscan.io/extrinsic/0x1ddbf6b06f6204ae1a2992bca766f4183d73b8e1fdf23815a1b3cbb29fcde35c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**